boot/boot_serial/src/boot_serial.c

```diff
@@ -257,102 +252,38 @@ bs_upload(char *buf, int len)
      * }
      */
 
-    /*
-     * Object comes within { ... }
-     */
-    cbor_buf_reader_init(&reader, (uint8_t *)buf, len);
-#ifdef __ZEPHYR__
-    cbor_parser_cust_reader_init(&reader.r, 0, &parser, &root_value);
-#else
-    cbor_parser_init(&reader.r, 0, &parser, &root_value);
-#endif
-    if (!cbor_value_is_container(&root_value)) {
+    Upload_t upload;
+    if (!cbor_decode_Upload((const uint8_t *)buf, len, &upload)) {
         goto out_invalid_data;
     }
-    if (cbor_value_enter_container(&root_value, &value)) {
+
+    if (upload._Upload_members_count > 5) {
```

Comment:
Instead of literal 5, can you use some named define? Why 5?

Reply:
It was because we have 5 different keys in the CBOR map ("image", "sha" etc.). I removed the checks altogether because it's not strictly needed, and will cause incompatibility with newer formats with additional keys.